### PR TITLE
[CLI-2132] Add Enterprise SKU support

### PR DESCRIPTION
**Pull request by sgagniere**
Release Notes
-------------

Breaking Changes
- PLACEHOLDER

New Features
- PLACEHOLDER

Bug Fixes
- PLACEHOLDER

Checklist
---------
1. [CRUCIAL] Is the change for features that are already live in prod?
   * yes: ok

What
----
Add esku support for `confluent kafka cluster` and `confluent price list` commands.

References
----------

Test & Review
-------------
New integration tests
Manual testing

Open Questions / Follow-ups
---------------------------

### CODE REVIEW

File: internal/pkg/errors/error_message.go [hunk 1]
```diff
@@ -145,7 +145,7 @@ const (
 	InvalidAvailableFlagErrorMsg                     = "invalid value \"%s\" for `--availability` flag"
 	InvalidAvailableFlagSuggestions                  = "Allowed values for `--availability` flag are: %s, %s."
 	InvalidTypeFlagErrorMsg                          = "invalid value \"%s\" for `--type` flag"
-	InvalidTypeFlagSuggestions                       = "Allowed values for `--type` flag are: %s, %s, %s."
+	InvalidTypeFlagSuggestions                       = "Allowed values for `--type` flag are: %s, %s, %s, %s."
```
Comment (brianstrauch) [inline internal/pkg/errors/error_message.go:148]: Looks like this is only being used in one place, so does it need to be global?

Comment (brianstrauch) [reply internal/pkg/errors/error_message.go:148]: Also, can we use one `%s` and insert the entire comma-delimited list so we don't have to keep updating this as more cluster types are added?

File: internal/pkg/errors/catcher.go [hunk 2]
```diff
@@ -259,7 +259,10 @@ func CatchClusterConfigurationNotValidError(err error, r *http.Response) error {
 
 	err = CatchCCloudV2Error(err, r)
 	if strings.Contains(err.Error(), "CKU must be greater") {
-		return New(InvalidCkuErrorMsg)
+		return New("CKU must be greater than 1 for multi-zone dedicated cluster")
+	}
+	if strings.Contains(err.Error(), "Durability must be HIGH for an Enterprise cluster") {
+		return New("availability must be `multi-zone` for enterprise cluster")
```
[the remark this answers is not in the document]
Comment (brianstrauch) [reply internal/pkg/errors/catcher.go:265]: FYI: Backticks are only used for commands and flags

[the remark this answers is not in the document]
Comment (sgagniere) [reply internal/pkg/errors/catcher.go:265]: That would be ideal. I'll put in a ticket afterwards.

File: internal/cmd/kafka/command_cluster_create.go [hunk 3]
```diff
@@ -264,7 +266,7 @@ func stringToSku(skuType string) (ccstructs.Sku, error) {
 		break
 	default:
 		return ccstructs.Sku_UNKNOWN, errors.NewErrorWithSuggestions(fmt.Sprintf(errors.InvalidTypeFlagErrorMsg, skuType),
-			fmt.Sprintf(errors.InvalidTypeFlagSuggestions, skuBasic, skuStandard, skuEnterprise, skuDedicated))
+			fmt.Sprintf("Allowed values for `--type` flag are: %s.", utils.ArrayToCommaDelimitedString(kafka.Types, "or")))
```
Comment (brianstrauch) [inline internal/cmd/kafka/command_cluster_create.go:269]:
```suggestion
			fmt.Sprintf("Allowed values for `--type` flag are: %s.", utils.ArrayToCommaDelimitedString(kafka.Types, "and")))
```

**Review by brianstrauch — APPROVED** [review]

**Pull request merged**